### PR TITLE
Remove global mutable config to allow for concurrency

### PR DESCRIPTION
**Pull request by marcusklaas**
This should make it possible to run idempotence/ system tests concurrently.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -14,7 +14,7 @@ use {NewlineStyle, BraceStyle, ReturnIndent};
 use lists::SeparatorTactic;
 use issues::ReportTactic;
 
-#[derive(RustcDecodable)]
+#[derive(RustcDecodable, Clone, Copy)]
```
Comment (nrc) [inline src/config.rs]: This shouldn't be `Copy` - it might get more complex in the future and is already too big to morally be copy-able

**Comment by nrc** [comment]
This is a good idea, the only change I would make is to not copy the config around (see the comment on deriving copy). That is going to require changing a few of the uses to references and will need a few lifetime params scattered around, but should work.

**Comment by marcusklaas** [comment]
It luckily was very little work!

**Comment by nrc** [comment]
Needs a rebase now

**Comment by marcusklaas** [comment]
Rebasified!

**Pull request merged**